### ISSUE TITLE
[BUG] Maven Dependency Management is ignored, so some dependencies are not downloaded

### ISSUE THREAD

**Issue by kmruiz** [kmruiz/pottery#3]
**Describe the bug**
Pottery is ignoring the dependencyManagement in maven dependencies, so the resolution of some packages is not working as expected, failing when generating the fatJar.

**Impact of the bug**
Some applications that require frameworks with transitives dependencies using dependencyManagement won't work.

* [x] I1: This doesn't allow us to release our software using pottery.
* [ ] I2: The feature is useful for us but doesn't block us.
* [ ] I3: Is a quality of life issue, it's a pain: would be nice to fix.

**To Reproduce**
1. Create a new pottery project with the following yaml:
```yaml
parameters:
  junit.version: "5.9.1"
  javalin.version: "5.3.0"

artifact:
  group:    "cat.pottery"
  id:       "javalin_example"
  version:  "0.0.1"

  platform:
    version: "17"
    produces:
      - "fatjar"

  manifest:
    main-class: "cat.pottery.ui.cli.Bootstrap"

  dependencies:
    - production: "io.javalin:javalin:${javalin.version}"
    - test: "org.junit.jupiter:junit-jupiter-api:${junit.version}"
```
2. Create a main class, it can be empty with a simple Hello World, named `cat.pottery.ui.cli.Bootstrap`
3. Run the package command.

**Expected behavior**
A fatJar with the application and kotlin.

**Screenshots**

**Logs**
```sh
Exception in thread "main" java.lang.Error: java.util.zip.ZipException: zip END header not found
	at cat.pottery.engine.output.fatJar.FatJarArtifactOutput.lambda$generateArtifact$2(FatJarArtifactOutput.java:93)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at cat.pottery.engine.output.fatJar.FatJarArtifactOutput.generateArtifact(FatJarArtifactOutput.java:67)
	at cat.pottery.ui.cli.command.PackageCommand.execute(PackageCommand.java:68)
	at cat.pottery.ui.cli.Bootstrap.main(Bootstrap.java:45)
Caused by: java.util.zip.ZipException: zip END header not found
	at java.base/java.util.zip.ZipFile$Source.findEND(ZipFile.java:1474)
	at java.base/java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1482)
	at java.base/java.util.zip.ZipFile$Source.<init>(ZipFile.java:1320)
	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1282)
	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:709)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:243)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:172)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:186)
	at cat.pottery.engine.output.fatJar.FatJarArtifactOutput.lambda$generateArtifact$2(FatJarArtifactOutput.java:73)
	... 4 more

Process finished with exit code 1
```

**Environment**
- OS: Any 
- JDK: Any
- Pottery Version: Any pottery version

**pottery.yaml**
```yaml
parameters:
  junit.version: "5.9.1"
  javalin.version: "5.3.0"

artifact:
  group:    "cat.pottery"
  id:       "javalin_example"
  version:  "0.0.1"

  platform:
    version: "17"
    produces:
      - "fatjar"

  manifest:
    main-class: "cat.pottery.ui.cli.Bootstrap"

  dependencies:
    - production: "io.javalin:javalin:${javalin.version}"
    - test: "org.junit.jupiter:junit-jupiter-api:${junit.version}"
```

**Additional Context**
Offending POM: https://repo1.maven.org/maven2/io/javalin/javalin-parent/5.3.0/javalin-parent-5.3.0.pom

**Comment by kmruiz** [comment]
There was an additional issue regarding creating the fatJar Class-Path.

Related to the JDK documentation, the Class-Path is only necessary when a JAR depends on other JARs, which is not necessary in a fatJar.

**Comment by kmruiz** [comment]
Releasing Fix on 0.3.3
